libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/UIDevice+SFHardware.m

```diff
@@ -187,119 +216,127 @@ - (NSNumber *)sfsdk_freeDiskSpace {
+        @"iPad13,16": @(UIDevice5GiPadAir),         // iPad Air (5th generation)
+        @"iPad13,17": @(UIDevice5GiPadAir),         // iPad Air (5th generation)
+        @"iPad13,18": @(UIDevice10GiPad),           // iPad (10th generation)
+        @"iPad13,19": @(UIDevice10GiPad),           // iPad (10th generation)
+        @"iPad14,1": @(UIDevice6GiPadMini),         // iPad mini (6th generation)
+        @"iPad14,2": @(UIDevice6GiPadMini),         // iPad mini (6th generation)
+        @"iPad14,3": @(UIDevice6G129InchiPadPro),   // iPad Pro (12.9-inch, 6th generation)
+        @"iPad14,4": @(UIDevice6G129InchiPadPro),   // iPad Pro (12.9-inch, 6th generation)
+        @"iPad14,5": @(UIDevice11Inch4GiPadPro),    // iPad Pro (11-inch, 4th generation)
+        @"iPad14,6": @(UIDevice11Inch4GiPadPro),    // iPad Pro (11-inch, 4th generation)
+        @"iPad14,8": @(UIDevice6GiPadAir),          // iPad Air 6th Gen
+        @"iPad14,9": @(UIDevice6GiPadAir),          // iPad Air 6th Gen
+        @"iPad14,10": @(UIDevice7GiPadAir),          //iPad Air 7th Gen
+        @"iPad14,11": @(UIDevice7GiPadAir),          //iPad Air 7th Gen
+        @"iPad16,1": @(UIDevice7GiPadMini),          //iPad mini 7th Gen (WiFi)
+        @"iPad16,2": @(UIDevice7GiPadMini),          //iPad mini 7th Gen (WiFi+Cellular)
+        @"iPad16,3": @(UIDevice11Inch5GiPadPro),     //iPad Pro 11 inch 5th Gen
+        @"iPad16,4": @(UIDevice11Inch5GiPadPro),     //iPad Pro 11 inch 5th Gen
+        @"iPad16,5": @(UIDevice12Inch7GiPadPro),     //iPad Pro 12.9 inch 7th Gen
+        @"iPad16,6": @(UIDevice12Inch7GiPadPro),     //iPad Pro 12.9 inch 7th Gen
+    };
+    NSNumber *ipadType = ipadIdentifiers[platform];
+    if (ipadType) return ipadType.integerValue;
     
-    if ([platform hasPrefix:@"iPod"])                   return UIDeviceUnknowniPod;
-    if ([platform hasPrefix:@"iPad"])                   return UIDeviceUnknowniPad;
-    if ([platform hasPrefix:@"AppleTV"])                return UIDeviceUnknownAppleTV;
+    // Apple TVs
+    NSDictionary *appleTVIdentifiers = @{
+        @"AppleTV5,3": @(UIDeviceAppleTV4),        // Apple TV HD
+        @"AppleTV6,2": @(UIDeviceAppleTV4k),       // Apple TV 4K (1st generation)
+        @"AppleTV11,1": @(UIDeviceAppleTV4k2G),    // Apple TV 4K (2nd generation)
+        @"AppleTV14,1": @(UIDeviceAppleTV4k3G),    // Apple TV 4K (3rd generation)
+    };
+    NSNumber *appleTVType = appleTVIdentifiers[platform];
+    if (appleTVType) return appleTVType.integerValue;
     
-    // Simulator thanks Jordan Breeding
-    #if !TARGET_OS_VISION
-        if ([platform hasSuffix:@"86"] || [platform isEqual:@"x86_64"]) {
-            BOOL smallerScreen = [[UIScreen mainScreen] bounds].size.width < 768;
-            return smallerScreen ? UIDeviceSimulatoriPhone : UIDeviceSimulatoriPad;
-        }
-    #endif
-
-    return UIDeviceUnknown;
+    return UIDeviceUnknown;  // Unknown platform fallback
```

Comment:
I was just testing the latest and I noticed that the old code used to have more unknown subcategories (snippet below), thinking we should check those too before returning `UIDeviceUnknown` 

```
if ([platform hasPrefix:@"iPhone"]) return UIDeviceUnknowniPhone;
if ([platform hasPrefix:@"iPad"]) return UIDeviceUnknowniPad;
```

Reply:
ah, so add these before we return straight unknow?  OK, I'll add that back to make sure I don't regress expected behaviour here.